std-bits/table/src/main/java/org/enso/table/data/column/builder/InferredBuilder.java

```diff
@@ -193,9 +171,8 @@ private void retypeAndAppend(Object o) {
 
   private void retypeToMixed() {
     // The new internal builder must be at least `currentSize` so it can store
-    // all the current values. It must also be at least 'initialSize' since the
-    // caller might be using appendNoGrow and is expecting to write at least
-    // that many values.
+    // all the current values. Make it the same as the `initialSize` to avoid
+    // unnecessary reallocation.
```

Comment:
The phrasing is not quite clear to me here.

What about?

```suggestion
    // all the current values. We make it at least `initialSize` to avoid
    // unnecessary reallocation.
```